addons/pos_loyalty/models/pos_session.py

```diff
@@ -90,7 +90,7 @@ def load_data(self, models_to_load, only_data=False):
         result = super().load_data(models_to_load, only_data)
 
         # adapt product
-        if len(models_to_load) == 0 or 'product.product' in models_to_load:
+        if not only_data and (len(models_to_load) == 0 or 'product.product' in models_to_load):
```

Comment:
Adapt your changes for below lines only: 

https://github.com/odoo/odoo/blob/c04f433475d70f3266c55929812865e6099850d4/addons/pos_loyalty/models/pos_session.py#L103-L106